File: 프로그래머스/lv2/42586. 기능개발/기능개발.java
import java.util.Queue;
import java.util.LinkedList;
import java.util.Arrays;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int n = speeds.length;
        int[] answer = new int[n];
        Queue<Integer> queue = new LinkedList<>();
        
        for(int i=0; i<n; i++){
            int day = (100-progresses[i])%speeds[i];
            day=(day>0)?(100-progresses[i])/speeds[i]+1:(100-progresses[i])/speeds[i];
            queue.offer(day);
        }
        
        for(int i=0; i<n; i++){
            if(queue.isEmpty()){
                answer = Arrays.copyOf(answer,i);
                break;
            }
            int day=queue.poll();
            int num=1;
            while(queue.peek()!=null && day >= queue.peek()){
                queue.poll();
                num++;
            }
            answer[i]=num;
        }
        return answer;
    }
}

File: 프로그래머스/lv2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <queue>
using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    queue<pair<int,int>> q;
    int num=0;
    
    for(int i=0; i<progresses.size(); i++){
        q.push({progresses[i],speeds[i]});
    }
    
    for(int time=1; q.size()!=0; time++){
        num=0;
        while(q.front().first + q.front().second*time >= 100){
            q.pop();
            num++;
        }
        if(num!=0)
            answer.push_back(num);
    }
    
    return answer;
}